pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -893,8 +894,9 @@ func listRegistrationEntries(tx *gorm.DB,
 		// fetch the entries in the id set, filtered by any parent/spiffe id filters
 		// applied globally
 		var models []RegisteredEntry
-		if err := entryTx.Where(entryIDs).Find(&models).Error; err != nil {
-			return nil, sqlError.Wrap(err)
+		db := entryTx.Where(entryIDs)
+		if err := findRegisteredEntries(req, db, &models); err != nil {
```

Comment:
i think this is less confusing with a functional approach:

```
models, pagination, err := findRegisteredEntries(db, req.Pagination)
if err != nil {
```

Reply:
done